programs/client/BuzzHouse/Generator/ExternalIntegrations.h

```diff
@@ -0,0 +1,953 @@
+#pragma once
+
+#include "config.h"
+
+#include "FuzzConfig.h"
+#include "SQLCatalog.h"
+
+#if defined USE_MYSQL && USE_MYSQL
+#    if __has_include(<mysql.h>)
+#        include <mysql.h>
+#    else
+#        include <mysql/mysql.h>
+#    endif
+#endif
+
+#if defined USE_MONGODB && USE_MONGODB
+#    include <bsoncxx/builder/stream/array.hpp>
+#    include <bsoncxx/builder/stream/document.hpp>
+#    include <bsoncxx/json.hpp>
+#    include <bsoncxx/types.hpp>
+#    include <mongocxx/client.hpp>
+#    include <mongocxx/collection.hpp>
+#    include <mongocxx/database.hpp>
+#endif
+
+#if defined USE_LIBPQXX && USE_LIBPQXX
+#    include <pqxx/pqxx>
+#endif
+
+#if defined USE_SQLITE && USE_SQLITE
+#    include <sqlite3.h>
+#endif
+
+
+namespace BuzzHouse
+{
+
+using IntegrationCall = enum IntegrationCall { IntMySQL = 1, IntPostgreSQL = 2, IntSQLite = 3, IntRedis = 4, IntMongoDB = 5, IntMinIO = 6 };
+
+class ClickHouseIntegration
+{
+public:
+    const FuzzConfig & fc;
+    const ServerCredentials & sc;
+    std::string buf;
+
+    ClickHouseIntegration(const FuzzConfig & fcc, const ServerCredentials & scc) : fc(fcc), sc(scc) { buf.reserve(4096); }
+
+    virtual void setEngineDetails(RandomGenerator &, const SQLBase &, const std::string &, TableEngine *) { }
+
+    virtual bool performIntegration(RandomGenerator &, uint32_t, bool, std::vector<ColumnPathChain> &) { return false; }
+
+    virtual ~ClickHouseIntegration() = default;
+};
+
+class ClickHouseIntegratedDatabase : public ClickHouseIntegration
+{
+public:
+    std::ofstream out_file;
+    explicit ClickHouseIntegratedDatabase(const FuzzConfig & fcc, const ServerCredentials & scc)
+        : ClickHouseIntegration(fcc, scc), out_file(std::ofstream(scc.query_log_file, std::ios::out | std::ios::trunc))
+    {
+    }
+
+    virtual bool performQuery(const std::string &) { return false; }
+
+    virtual std::string getTableName(uint32_t) { return std::string(); }
+
+    virtual void getTypeString(RandomGenerator &, SQLType *, std::string &) const { }
```

Comment:
I'd expect from any `get` method to return something. Seems you're returning the result in the 3rd parameter? I've seen that throughout the code, but I don't understand why having an out parameter is any better than returning the value directly for your use cases.

Reply:
avoid allocating strings, that's it